nullaway/src/main/java/com/uber/nullaway/Nullness.java

```diff
@@ -127,26 +128,41 @@ public String toString() {
     return displayName;
   }
 
-  private static Nullness nullnessFromAnnotations(Element element) {
-    for (AnnotationMirror anno : NullabilityUtil.getAllAnnotations(element)) {
-      String annotStr = anno.getAnnotationType().toString();
-      if (isNullableAnnotation(annotStr)) {
-        return Nullness.NULLABLE;
+  public static final class AnnotationReader {
+
+    private Set<String> nullableAnnotations;
+
+    public AnnotationReader(Config config) {
+      nullableAnnotations = config.getNullableAnnotations();
+    }
+
+    private Nullness nullnessFromAnnotations(Element element) {
+      for (AnnotationMirror anno : NullabilityUtil.getAllAnnotations(element)) {
+        String annotStr = anno.getAnnotationType().toString();
+        if (isNullableAnnotation(annotStr)) {
+          return Nullness.NULLABLE;
+        }
       }
+      return Nullness.NONNULL;
     }
-    return Nullness.NONNULL;
-  }
 
-  /**
-   * @param annotName annotation name
-   * @return true if we treat annotName as a <code>@Nullable</code> annotation, false otherwise
-   */
-  public static boolean isNullableAnnotation(String annotName) {
-    return annotName.endsWith(".Nullable")
-        || annotName.equals("org.checkerframework.checker.nullness.compatqual.NullableDecl");
-  }
+    /**
+     * @param annotName annotation name
+     * @return true if we treat annotName as a <code>@Nullable</code> annotation, false otherwise
+     */
+    public boolean isNullableAnnotation(String annotName) {
```

Comment:
you could potentially memoize the result of this as well with a reasonably sized lru cache so you can avoid looking up the same annotation multiple times

Reply:
How much faster do we expect the dictionary lookup to be compared to iterating over a set of `n~=5`? I don't think there will ever be a huge set of `@Nullable` equivalent annotations here. That said, we could benchmark both approaches, for sure!

Reply:
We are looking at all instances of annotations used in the source code

Everytime you lookup an annotation, you keep a bit about whether it is nullable or not.

The source code can refer to the same annotation in many places so it saves us some time not needing to compare against an ever growing list of nullable annotations